app/javascript/components/organisation-selector.js

```diff
@@ -0,0 +1,27 @@
+class OrganisationSelector {
+  constructor() {
+    this.selectElt = document.querySelector(".js-organisation-selector");
+    if (this.selectElt === null) return;
+
+    this.attachListener();
+  }
+
+  attachListener() {
+    this.selectElt.addEventListener("change", (event) => {
+      this.refreshQuery(event.target.value);
+    });
+  }
+
+  refreshQuery(selectedOrganisation) {
+    if (selectedOrganisation && selectedOrganisation !== "0") {
+      const url = new URL(`${window.location.origin}/organisations/${selectedOrganisation}/stats`);
+      window.location.href = url;
+    } else {
+      const currentDepartmentId = document.getElementById("department_id").value
+      const url = new URL(`${window.location.origin}/departments/${currentDepartmentId}/stats`);
+      window.location.href = url;
+    }
+  }
```

Comment:
Peut-être un peu mieux comme ça :
```javascript
if (selectedOrganisation && selectedOrganisation !== "0") {
  const url = new URL(`${window.location.origin}/organisations/${selectedOrganisation}/stats`);
} else {
  const currentDepartmentId = document.getElementById("department_id").value
  const url = new URL(`${window.location.origin}/departments/${currentDepartmentId}/stats`);
}
window.location.href = url;
```

Reply:
`url` n'est pas défini dans ce scope avec cette syntaxe (damn you javascript). Mais tu as raison la ligne doublonée n'est pas géniale, j'ai modifié en déinissant la variable dans un `let` avant le `if` ; ça fait pas gagner de ligne mais c'est plus DRY